Tasks/AzureRmWebAppDeploymentV4/operations/KuduServiceUtility.ts

```diff
@@ -158,6 +158,12 @@ export class KuduServiceUtility {
                 await webClient.sleepFor(5);
             }
 
+            if(tl.stats(packagePath).isDirectory()) {
+                let tempPackagePath = deployUtility.generateTemporaryFolderOrZipPath(tl.getVariable('AGENT.TEMPDIRECTORY'), false);
+                packagePath = await zipUtility.archiveFolder(packagePath, "", tempPackagePath);
+                tl.debug("Compressed folder " + packagePath + " into zip : " +  packagePath);
+            }
+
```

Comment:
Can we move this logic inside provider ?

Reply:
As Discussed you can move this logic in separate PR. As in existing Provider layer we are checking if it is folder don't zip it but in next layer Kudu service API call we are again checking if folder zip this folder.

Reply:
thanks, moved the logic to deploymentProvider.